src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1510,4 +1510,25 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
+  <data name="PreferDictionaryContainsKeyCodeFixTitle" xml:space="preserve">
+    <value>Use 'ContainsKey(TKey)'</value>
+  </data>
+  <data name="PreferDictionaryContainsKeyDescription" xml:space="preserve">
+    <value>'{0}.ContainsKey(TKey)' is usually O(1), while '{0}.Keys.Contains(TKey)' may be O(n) in some cases. Additionally, many dictionary implementations lazily initialize the Keys collection to cut back on allocations.</value>
+  </data>
+  <data name="PreferDictionaryContainsKeyMessage" xml:space="preserve">
+    <value>Prefer '{0}.ContainsKey(TKey)' over '{0}.Keys.Contains(TKey)'</value>
```

Comment:
Below might be more readable:
```suggestion
    <value>Prefer 'ContainsKey' over 'Keys.Contains' for dictionary type '{0}'</value>
```

Reply:
I agree. That looks much better. I'll change it.